pkg/provider/storage/azure_storageaccount.go

```diff
@@ -532,56 +604,70 @@ func (az *Cloud) EnsureStorageAccount(ctx context.Context, accountOptions *Accou
 				enableBlobVersioning = ptr.To(*accountOptions.EnableBlobVersioning)
 			}
 
-			property := storage.BlobServiceProperties{
-				BlobServicePropertiesProperties: &storage.BlobServicePropertiesProperties{
+			property := armstorage.BlobServiceProperties{
+				BlobServiceProperties: &armstorage.BlobServicePropertiesProperties{
 					IsVersioningEnabled:            enableBlobVersioning,
 					ContainerDeleteRetentionPolicy: containerPolicy,
 					DeleteRetentionPolicy:          blobPolicy,
 				},
 			}
-
-			if _, err := az.BlobClient.SetServiceProperties(ctx, subsID, resourceGroup, accountName, property); err != nil {
+			var blobserviceClient blobservicepropertiesclient.Interface
+			if accountOptions.SubscriptionID != "" {
+				blobserviceClient, err = clientFactory.GetBlobServicePropertiesClientForSub(subsID)
+				if err != nil {
+					return "", "", fmt.Errorf("failed to get blob service properties client, error: %w", err)
+				}
+			} else {
+				blobserviceClient = clientFactory.GetBlobServicePropertiesClient()
```

Comment:
same here, we should only use `GetBlobServicePropertiesClientForSub(subsID)`, right?

Reply:
fixed.